src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/CreateSdkSymbolsLayout.cs

```diff
@@ -0,0 +1,165 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    // Creates a symbols layout that matches the SDK layout
+    public class CreateSdkSymbolsLayout : Task
+    {
+        private Hashtable AllPdbGuids;
+
+        /// <summary>
+        /// Path to SDK layout.
+        /// </summary>
+        [Required]
+        public string SdkLayoutPath { get; set; }
+
+        /// <summary>
+        /// Path to all source-built symbols, flat or with folder hierarchy.
+        /// </summary>
+        [Required]
+        public string AllSymbolsPath { get; set; }
+
+        /// <summary>
+        /// Path to SDK symbols layout - will be created if it doesn't exist.
+        /// </summary>
+        [Required]
+        public string SdkSymbolsLayoutPath { get; set; }
+
+        /// <summary>
+        /// If true, fails the build if any PDBs are missing.
+        /// </summary>
+        public bool FailOnMissingPDBs { get; set; }
+
+        public override bool Execute()
+        {
+            IndexAllSymbols(AllSymbolsPath);
```

Comment:
The parameter isn't needed here - will fix.

Reply:
Fixed in https://github.com/dotnet/installer/pull/17454/commits/b9f64fd009e8a3be9ad733950302f7cf7d963116